Subject: Allow running cron in foreground
Last-Update: 2010-04-11

Allow running cron in foreground if one wishes to do so.

Bug-Debian: http://bugs.debian.org/108492

Acked-by: Christian Kastner <debian@kvr.at>
Index: patched/cron.c
===================================================================
--- patched.orig/cron.c	2010-05-06 18:18:54.665561020 +0200
+++ patched/cron.c	2010-05-06 18:18:55.651562056 +0200
@@ -117,7 +117,7 @@
 	if (0) {
 # endif
 		(void) fprintf(stderr, "[%d] cron started\n", getpid());
-	} else {
+	} else if (!stay_foreground) {
 		switch (fork()) {
 		case -1:
 			log_it("CRON",getpid(),"DEATH","can't fork");
@@ -464,12 +464,16 @@
 {
 	int	argch;
 
+	stay_foreground = 0;
         lsbsysinit_mode = 0;
 
-	while (EOF != (argch = getopt(argc, argv, "lx:"))) {
+	while (EOF != (argch = getopt(argc, argv, "lfx:"))) {
 		switch (argch) {
 		default:
 			usage();
+		case 'f':
+			stay_foreground = 1;
+			break;
 		case 'x':
 			if (!set_debug_flags(optarg))
 				usage();
Index: patched/cron.h
===================================================================
--- patched.orig/cron.h	2010-05-06 18:18:54.665561020 +0200
+++ patched/cron.h	2010-05-06 18:18:55.651562056 +0200
@@ -290,6 +290,7 @@
 time_min clockTime;
 static long GMToff;
 
+int	stay_foreground;
 int     lsbsysinit_mode;
 
 char    cron_default_mail_charset[MAX_ENVSTR] = "";
